trader.py

```diff
@@ -1,99 +1,246 @@
-            asks = sorted(order_depth.sell_orders.keys())
-            for ask in asks:
-                if ask < fair_value:
-                    volume = order_depth.sell_orders[ask]
-                    orders.append(Order(product, ask, -volume))
-        if order_depth.buy_orders and position > -limit:
-            bids = sorted(order_depth.buy_orders.keys(), reverse=True)
-            for bid in bids:
-                if bid > fair_value:
-                    volume = order_depth.buy_orders[bid]
-                    orders.append(Order(product, bid, -volume))
-        return orders
-
-    def trade_berries(self, state: TradingState) -> List[Order]:
-        product = "BERRIES"
-        limit = 250
-        order_depth, position, timestamp = state.order_depths.get(
-            product, None), state.position.get(product, 0), state.timestamp
-        if not order_depth:
-            return []
-
-        # # backtesting
-        # buy_ends_at = 30 * 1000
-        # sell_starts_at = 50 * 1000
-
-        # production
-        buy_ends_at = 300 * 1000
-        sell_starts_at = 500 * 1000
-
-        if timestamp < buy_ends_at and order_depth.sell_orders and position < limit:
-            best_ask = min(order_depth.sell_orders.keys())
-            volume = order_depth.sell_orders[best_ask]
-            return [Order(product, best_ask, -volume)]
-        if timestamp > sell_starts_at and order_depth.buy_orders and position > -limit:
-            best_bid = max(order_depth.buy_orders.keys())
-            volume = order_depth.buy_orders[best_bid]
-            return [Order(product, best_bid, -volume)]
-        return []
+import json
+from typing import Dict, List, Any
+from datamodel import *
+from collections import defaultdict
+import numpy as np
+
+class Logger:
+    # Set this to true, if u want to create
+    # local logs
+    local: bool
+    # this is used as a buffer for logs
+    # instead of stdout
+    local_logs: dict[int, str] = {}
+
+    def __init__(self, local=False) -> None:
+        self.logs = ""
+        self.local = local
+
+
+    def print(self, *objects: Any, sep: str = " ", end: str = "\n") -> None:
+        self.logs += sep.join(map(str, objects)) + end
+
+    def flush(self, state: TradingState, orders: dict[Symbol, list[Order]]) -> None:
+        output = json.dumps({
+            "state": state,
+            "orders": orders,
+            "logs": self.logs,
+        }, cls=ProsperityEncoder, separators=(",", ":"), sort_keys=True)
+        if self.local:
+            self.local_logs[state.timestamp] = output
+        print(output)
+
+        self.logs = ""
+
+
+class Trader:
+
+    logger = Logger(local=True)
+
+    def __init__(self) :
+        self.trade_prices = defaultdict(list)
+        self.logs = ""
+        self.pos_limit = {"PEARLS": 20, "BANANAS": 20, "COCONUTS": 600,
+                          "PINA_COLADAS": 300, "BERRIES": 250, "DIVING_GEAR": 50}
+        self.pos = {"PEARLS": 0, "BANANAS": 0, "COCONUTS": 0,

```

Comment:
`state.position` would give you this which would be better than tracking it yourself?